### PR TITLE
wip: badge-type details

### PR DESCRIPTION
**Pull request by nicosampler**

### CODE REVIEW

**Comment by nicosampler** [comment]
@FedeMadoery I need your help here.

I just realized that we have a little problem with the component that creates the kleros input for the form. the final value after selecting the input type is saving the KEY instead of the value.

This is an example of the metadata we are saving to IPFS:
image


This is an example of how TCR does it:
image

If you can, could you please do a fix in this PR?

**Comment by FedeMadoery** [comment]
@nicosampler As a friendly reminder, the queries are not working as expected...

```
⚠ Generate outputs
  ❯ Generate to types/generated/subgraph.ts
    ✔ Load GraphQL schemas
    ✔ Load GraphQL documents
    ✖ GraphQL Document Validation failed with 4 errors;
      Error 0: Cannot query field "emitter" on type "BadgeType".
      at ../src/subgraph/queries/badgeType.ts:11:7
      Error 1: Cannot query field "emitter" on type "BadgeType".
      at ..src/subgraph/queries/badgeType.ts:10:7
      Error 2: Cannot query field "emitters" on type "Query".
      at ../src/subgraph/queries/creators.ts:3:5
      Error 3: Cannot query field "emitter" on type "Query".
      at ../src/subgraph/queries/creators.ts:3:5
```

File: pages/badge/mint/[typeId].tsx [hunk 1]
```diff
@@ -1,69 +1,102 @@
...
+  // TODO: hardcoded for now, as we only support Kleros.
+  const badgeTypeMetadata = useS3Metadata<{ file: klerosListStructure }>(
+    badgeType.data?.badgeType?.klerosBadge?.klerosMetadataURL || '',
+  )
+  if (badgeTypeMetadata.error || !badgeTypeMetadata.data) {
+    throw `There was an error trying to fetch the metadata for the badge type`
+  }
+
+  const CreateBadgeSchema = z.object({
+    evidenceFileUri: FileSchema.describe(
```
Comment (FedeMadoery) [inline pages/badge/mint/[typeId].tsx:54]: Is this field temporal? It is supposed to be requested by the creator, so it should be on the `columns`

File: pages/badge/mint/[typeId].tsx [hunk 2]
```diff
@@ -1,69 +1,102 @@
...
+    'KlerosBadgeTypeController',
+  )
+  const calls = [klerosController.badgeRequestValue] as const
+
+  const [{ data: klerosDepositCostData }] = useContractCall<
+    KlerosBadgeTypeController,
+    typeof calls
+  >(calls, [[badgeTypeId]], `klerosBadgeRequestValue-${badgeTypeId}`)
+
+  const klerosCost = klerosDepositCostData?.[0] || ZERO_BN
```
Comment (FedeMadoery) [inline pages/badge/mint/[typeId].tsx:71]: We may need to show an error if the `klerosDepositCost` is 0, right? Not now, that's for sure, but in a near future... So if we can add the comment and create a task, that would be great

File: pages/badge/mint/[typeId].tsx (hunk 1 above)
Comment (nicosampler) [reply pages/badge/mint/[typeId].tsx:54]: you are right! not sure why I've added it, but I will remove it. 
good catch!

File: pages/badge/mint/[typeId].tsx (hunk 2 above)
Comment (nicosampler) [reply pages/badge/mint/[typeId].tsx:71]: yes! totally, let me add a `throw` for now

**Review by FedeMadoery — APPROVED** [review]

**Pull request merged**